internal/colorize/cropped.go

```diff
@@ -0,0 +1,58 @@
+package colorize
+
+type Entry struct {
+	Line   string
+	Length int
+}
+
+func GetCroppedText(text string, maxLen int) []Entry {
+	entries := make([]Entry, 0)
+	entryText := ""
+	pos := 0
+	count := 0
+	matches := colorRx.FindAllSubmatchIndex([]byte(text), -1)
+	runeText := []rune(text)
+
+	for pos < len(runeText) {
+		// If we reach an index that we recognize (ie. the start of a tag)
+		// then we write the whole tag, otherwise write by rune
+		for _, match := range matches {
+			start, stop := match[0], match[1]
+			if pos == start {
+				entryText += string(runeText[pos:stop])
+				pos = stop
+			}
+		}
+
+		if pos > len(runeText)-1 {
```

Comment:
The for loop already handles this

Reply:
This check is from the matching that happens above. If `stop` (the of the reset tag) is at the end of the string then the index is out of bounds